### ISSUE TITLE
Polo 2008 VDO V06 Cluster GetSKC failure

### ISSUE THREAD

**Issue by YouGoKev** [gmenounos/kw1281test#62]
Hi Greg, firstly good job on your excellent work.

I tried to get SKC for my 2008 Polo 9N using the command KW1281 Com4 10400 17 GetSKC.  It seemed to connect OK however it was unable to unlock the cluster - please refer to log files attached.  Also attached is a dump of the EPROM but I've changed the file type from .bin to .log to enable it to be uploaded.  I don't understand how it got the dump if it was unable to unlock the cluster but most likely that is just ignorance on my part.

Can the SKC be extracted from the dump?  I'd appreciate instructions on how to do it if possible.

Hope the above and attached files help to refine your project.

Regards, Kevin
![KW1281Test 09MAR24](https://github.com/gmenounos/kw1281test/assets/162859423/2b811330-5e99-438a-957c-101c88ac9a47)
[KW1281Test.log](https://github.com/gmenounos/kw1281test/files/14549026/KW1281Test.log)
[6Q0920804Q__KOMBIINSTRUMENT_VDO_V06_$0090_eeprom.log](https://github.com/gmenounos/kw1281test/files/14549027/6Q0920804Q__KOMBIINSTRUMENT_VDO_V06_.0090_eeprom.log)

**Comment by YouGoKev** [comment]
Hi Greg.
I tried the 0.93 Beta as you recommended.  Still no joy.  The log file is attached.

Thanks for your assistance.

Regards, Kevin
[KW1281Test.log](https://github.com/gmenounos/kw1281test/files/14635897/KW1281Test.log)

**Comment by gmenounos** [comment]
Looks like you typed "Get SKC" instead of "GetSKC". Try it again without the space.

**Comment by YouGoKev** [comment]
Doh!

**Comment by YouGoKev** [comment]
Hi Greg
I tried 0.93 Beta - again no SKC offered unfortunately.  The log and bin files are attached as well as a screenshot of the Command Prompt screen - hopefully they offer some insight to what's going on.
Once again thanks for your effort on this.
Regards, Kevin
![KW1281Test_log_29MAR](https://github.com/gmenounos/kw1281test/assets/162859423/981c278c-2f2c-4865-8913-80243808f859)
[KW1281Test.log](https://github.com/gmenounos/kw1281test/files/14807056/KW1281Test.log)
[6Q0920804Q__KOMBIINSTRUMENT_VDO_V06_$0090_eeprom.txt](https://github.com/gmenounos/kw1281test/files/14807061/6Q0920804Q__KOMBIINSTRUMENT_VDO_V06_.0090_eeprom.txt)

**Comment by gmenounos** [comment]
Can you try this [new release](https://github.com/gmenounos/kw1281test/releases/tag/v0.97-beta) with your cluster and see if GetSKC works now?

However, the first thing you should do is use the kw1281test _reset_ command so that your cluster reboots. The newer clusters use a new secret key and if the old key had been used, the cluster will refuse to accept the new key until it is reset.

**Comment by YouGoKev** [comment]
Hi Greg
I'll try the new release.  Just so I understand correctly, do you want me to do a Reset command before I do the GetSKC command to retrieve the SKC OR do a Reset command before I try to use the retrieved SKC code to unlock the cluster?
Thanks, Kevin

**Comment by gmenounos** [comment]
Just before you use the GetSKC command.

**Comment by YouGoKev** [comment]
Hi Greg
Finally got access to the car again.  The Reset before doing the GetSKC did the trick - so success!  Well done you!  Would the dump file be of any use to you - if so, I'll post it up.
Using VCDSLite, when I attempt to login to the cluster using the retrieved SKC code I get an error saying "Login Lost due to dropped session".  Any clues as to what this means?
Thanks Kevin

**Comment by gmenounos** [comment]
Try connecting to the immobilizer (address 25) instead of the cluster (address 17) with VCDS and see if you can login with the SKC.

**Comment by YouGoKev** [comment]
Hi Greg, thanks for your reply
I was able to log into the immobilizer (25) OK, no error message displayed.  So it looks as though the SKC code is correct.

I've noticed that when I turn the ignition key on, the work "LOCKED" appears on the MFD where the transmission selection (PNRD12) is normally displayed - refer attached image. ![PXL_20240602_025744153](https://github.com/gmenounos/kw1281test/assets/162859423/39fe9ab5-00e3-4cc8-a841-3473da7b2f16) 
When I put my foot on the brake pedal the word "LOCKED" disappears.  I've not noticed this before - it could be "normal" behavior as a prompt to tell the driver that they need to put their foot on the brake pedal or it could be that the cluster is locked.
Do you have any recommendations?
Thanks for your assistance.
Regards, Kevin

**Comment by YouGoKev** [comment]
Hi Greg
Apparently this "LOCKED" indication is normal behavior - it's to indicate that the gear selection lever is locked (can't be moved from the PARK position) until you put your foot on the brake pedal - makes sense.

**Comment by gmenounos** [comment]
Ah, ok. I'm in the US and they don't sell Polos here, which is probably why I didn't have much luck researching that message. Glad you got it figured out.

**Comment by YouGoKev** [comment]
Hi Greg
Do you have any idea why I can't log into the cluster?
Thanks Kevin

**Comment by gmenounos** [comment]
Some clusters, like yours, have 2 addresses (17 for the cluster itself and 25 for the immobilizer). Those only let you log in to address 25 when adding new keys.

**Comment by YouGoKev** [comment]
Hi Greg
OK, thanks - I'll give it a go.
Thanks for your help once again.
Cheers, Kevin

**Comment by YouGoKev** [comment]
Hi Greg - just an update.  Yes, as you suggested I had to get into the immobiliser module to access the key coding.  I was able to successfully code the new key using VCDS-Lite but I needed an activated version to access that particular feature.
Again, thanks for your good work on the code and your advice & assistance.
I've made a small donation to the cause via PayPal. 
Hopefully, I won't have the need to be bothering you in the near future - but never say never:) 
Cheers for now.
Kevin
